src/main/java/org/apache/sling/jcr/packageinit/impl/ExecutionPlanRepoInitializer.java

```diff
@@ -157,9 +156,12 @@ public void processRepository(SlingRepository slingRepository) throws Exception
                             logger.info("No tasks found in execution plan - no additional packages installed.");
                         }
                         
-                        // save hashes to file for crosscheck on subsequent startup to avoid double processing
-                        writer.write(String.valueOf(plan.hashCode()));
-                        writer.newLine();
+                        if ( !hasSnapshot ) {
+                            // save hashes to file for crosscheck on subsequent startup to avoid double processing
+                            // but always process SNAPSHOT packages
+                            writer.write(String.valueOf(plan.hashCode()));
+                            writer.newLine();
+                        }
```

Comment:
Does this mean that _all_ packages are reinstalled if a snapshot package is part of the execution plan?

Reply:
Yes. The alternative would be to generate separate execution plans but that is too invasive. Would it make sense to log a WARN here if we have mixed SNAPSHOT and release packages?

Reply:
I don't know. I am not sufficiently familiar with how execution plans work. I am also not opposed to having a solution where all packages are reinstalled in this case. It _should_ not matter, because installing packages without snapshot qualifier should be a no-op. Of course, this is where hard to diagnose problems can be born. 🙂